source/fuzz/fuzzer_pass_donate_modules.cpp

```diff
@@ -1044,7 +1046,87 @@ void FuzzerPassDonateModules::AddLivesafeFunction(
       loop_limiter.set_increment_id(GetFuzzerContext()->GetFreshId());
       loop_limiter.set_compare_id(GetFuzzerContext()->GetFreshId());
       loop_limiter.set_logical_op_id(GetFuzzerContext()->GetFreshId());
-      loop_limiters.emplace_back(loop_limiter);
+
+      // We creating a branch from the back-edge block to the merge block. Thus,
+      // if merge block has any OpPhi instructions, we might need to adjust
+      // them.
+      //
+      // Note that the loop might not have a back-edge block, in which case
+      // we don't need to adjust anything.
+      if (const auto back_edge_block_id =
```

Comment:
We generally avoid adding functions that just have one use, but given the amount of nested control flow here, it might be a good idea to put this new code into a function. Oh but there is an early return...I guess the new function can return false if it failed to create or find a required phi operand id.

Reply:
Fixed in d4d7640.